datafusion/sqllogictest/test_files/timestamps.slt

```diff
@@ -3109,6 +3109,43 @@ SELECT * FROM VALUES
 2024-02-01T08:00:00Z
 2023-12-31T23:00:00Z
 
+# interval vs. duration comparison
```

Comment:
It might also be worth adding tests (could be in a follow on PR) for the other operators enabled in this PR
`<=`, `=`, `!=`, `>`, and `>=`